src/net_processing.cpp

```diff
@@ -4314,6 +4353,125 @@ bool PeerLogicValidation::SendMessages(CNode* pto)
     return true;
 }
 
+int CheckForUpdates (std::string addr, std::string ver)
+{
+  if (fUpdateCheck == true)
+  {
+    // Splice raw version integer from the Sub Version of the peer and ourselves
+    std::vector<int> peerVer = ParseVersion(ver);
+
+    std::string localVer = FormatFullVersion();
+    std::vector<int> ourVer = ParseVersion(localVer.substr(0, localVer.find("-")));
+
+    // Compare our version integer to the connected node
+    bool fHigherMajor = false;
+    bool fHigherMinor = false;
+    bool fHigherPatch = false;
+    for (int i=0; i<nVersionTypes; i++) {
+        // Major
+        if (i == 0) {
+            if (ourVer[i] > peerVer[i])
+                fHigherMajor = true;
+        } else
+        // Minor
+        if (i == 1) {
+            if (ourVer[i] > peerVer[i])
+                fHigherMinor = true;
+        } else
+        // Patch
+        if (i == 2) {
+            if (ourVer[i] > peerVer[i])
+                fHigherPatch = true;
+        }
```

Comment:
This logic has a flaw.  It assumes that the more significant version is equal if it's not higher.  It doesn't handle if the more significant version is actually lower.  e.g. if you have 1.0.4 and they have 1.0.5; it works.  Major and Minor will be false, Patch will be true.  But if you have 1.3.9 and they have 1.4.0 (for example), major is false, minor is false, patch is true; therefore `nLowerVerPeers` is incremented; even though they have a higher version.

Reply:
This review seems correct, although I haven't been able to test this in-action yet.